src/render.js

```diff
@@ -1,30 +1,38 @@
 const error = require('./error.js');
 const Exp = require('./expression.js');
 const assert = require('assert');
+const Promise = require('any-promise');
 
 var render = {
 
     renderTemplates: function(templates, scope) {
         assert(scope, 'unable to evalTemplates: scope undefined');
-        var html = '',
-            partial;
-        templates.some(template => {
+        var htmlBlocks = [],
+            partial,
+            promises = [];
+        templates.some((template, index) => {
```

Comment:
Be careful not let scope variables evaluated asyncly, which makes the implementation of control flow like `for` and `if` trivial. Note that async fashion may import implementation issues that does not exist in sync fashion:

What if the client user `{%capture%}` a variable (which invokes `renderTemplate`) and use it as `{%if%}` condition? We may need the promises resolved in serial in this situation.